### PR TITLE
Python3 units

### PR DESCRIPTION
**Pull request by rieder**
Conversion of units/

### CODE REVIEW

File: src/amuse/units/core.py [hunk 1]
```diff
@@ -978,8 +975,7 @@ def factor(self):
     @late
     def base(self):
         return tuple(
-            filter(lambda x: x[0] != 0,
-            map(lambda x: (x[0] + x[1], x[2]),self.combine_bases(self.left_hand.base, self.right_hand.base))))
+            [x for x in [(x[0] + x[1], x[2]) for x in self.combine_bases(self.left_hand.base, self.right_hand.base)] if x[0] != 0])
```
Comment (ipelupessy) [inline src/amuse/units/core.py]: for me this one needs parentheses...

File: src/amuse/units/core.py [hunk 2]
```diff
@@ -1046,8 +1042,7 @@ def reference_string(self):
     @late
     def base(self):
         return tuple(
-            filter(lambda x: x[0] != 0,
-            map(lambda x : (x[0] * self.power, x[1]), self.local_unit.base)))
+            [x for x in [(x[0] * self.power, x[1]) for x in self.local_unit.base] if x[0] != 0])
```
Comment (ipelupessy) [inline src/amuse/units/core.py]: also needs parentheses to be human readable

File: src/amuse/units/core.py [hunk 3]
```diff
@@ -1103,9 +1098,7 @@ def factor(self):
     @late
     def base(self):
         return tuple(
-                    filter(lambda x: x[0] != 0,
-                    map(lambda x: (x[0] - x[1], x[2]),
-                        self.combine_bases(self.left_hand.base, self.right_hand.base))))
+                    [x for x in [(x[0] - x[1], x[2]) for x in self.combine_bases(self.left_hand.base, self.right_hand.base)] if x[0] != 0])
```
Comment (ipelupessy) [inline src/amuse/units/core.py]: ..and same here

File: src/amuse/units/core.py (hunk 1 above)
Comment (rieder) [reply src/amuse/units/core.py]: fixed

File: src/amuse/units/core.py (hunk 2 above)
Comment (rieder) [reply src/amuse/units/core.py]: fixed

File: src/amuse/units/core.py (hunk 3 above)
Comment (rieder) [reply src/amuse/units/core.py]: fixed

**Pull request merged**